include/seastar/core/io_result.hh

```diff
@@ -0,0 +1,31 @@
+/*
+ * This file is open source software, licensed to you under the terms
+ * of the Apache License, Version 2.0 (the "License").  See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.  You may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/*
+ * Copyright 2022 Jinyong Ha (jyha200@gmail.com), Heewon Shin (shw096@snu.ac.kr)
+ */
+
+#pragma once
+
+namespace seastar {
+
+struct io_result {
```

Comment:
disclaimer: I'm not very happy with making the whole IO stack return `io-result` instead of `size_t`, but have no better suggestion :(
Patching IO stack with `io_result` support should come as a separate patch